types/jasmine/index.d.ts

```diff
@@ -15,7 +15,9 @@
 // TypeScript Version: 2.8
 // For ddescribe / iit use : https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/karma-jasmine/karma-jasmine.d.ts
 
-type ImplementationCallback = (() => Promise<any>) | ((done: DoneFn) => void);
+type PromiseOrPromiseLike<T> = Promise<T> | PromiseLike<T>;
```

Comment:
`Promise` implements `PromiseLike`, you can get rid of this type union and just use `PromiseLike`.

Reply:
Done, refactored everything to either return a PromiseLike or expect a PromiseLike as parameter. Had to add `await-promise: [true, "PromiseLike"]` to tslint options so that it will not fail.